### DISCUSSION TITLE
midi out from Host MIDI as AU in Logic

### DISCUSSION THREAD

**Discussion by pippux** [DISTRHO/Cardinal#318] [Q&A]
I cannot seem to get MIDI out from Host MIDI in the AU version in Logic. I think it is normal that AU do not output MIDI within logic itself but is there a way to get MIDI through the IAC driver? Is there a help page about this somewhere? 
Thank you
  F

Cardinal 22.07 (it says 22.06 in the help tab, I guess that is not correct)
Logic 10.5.1
Mac 10.14.6

**Comment by dromer** [comment]
22.07 has been released so anyway update please :)

(also this should probably be converted to an issue. or marked as duplicate if an issue for this already exists)

**Reply by pippux** [reply]
Oh, the package says 22.07 indeed, but the help tab says "Cardinal Unknown 22.06". I will edit the main post. 
Also, I would be very happy to post it as an issue, it's just that I am not 100% sure how it is supposed to work. Is it actually supposed to send MIDI to AIC? Since there is no AU for VCV I cannot rely on help from that community either.
One thing I will try to do is load the VST version in Kushview Elements and see if that works.

**Reply by falkTX** [reply]
oh I think I forgot to update the version on the AU side again.

it seems macos-intel package, targetting macOS 10.8, generates issues with MIDI.
the macos-universal package, which targets 10.10, is reportedly working fine when it comes to MIDI.

**Reply by pippux** [reply]
Thank you so much! I will try to install that version.

**Reply by pippux** [reply]
I tried with the universal package, still no luck.
But on a more fundamental level, it is even possible to output MIDI from Cardinal in Logic? 
 Did anyone try?  Probably I am really doing something wrong or maybe it is not supposed to work.
Thanks for the answers I got until now.

**Reply by falkTX** [reply]
I never tried this to be honest (do not have Logic, only GarageBand) but the code that makes it work is all there.
MIDI output from Cardinal does work, whether that works in AU mode is a good question.
Let me try quickly in FLStudio...

**Reply by falkTX** [reply]
ok indeed this does not work.
why exactly I couldnt find out. there are no errors being logged. seems to affect AU only, tested VST3 version and MIDI out works with that one.

**Reply by pippux** [reply]
Thanks again for the quick reply.
At this point I am starting to think it is a  Logic/AU limitation, I wouldn't be surprised.
The rest of the plugin is a terrific piece of work, I love it! Thanks for the great work!

**Reply by falkTX** [reply]
AudioUnits can do MIDI output, no? To be quite honest I dont actually know myself, as I do not have the need for AU plugins.
Until we know for sure it cannot be done (which I very much doubt it), we can just consider this a Cardinal-side bug.
